beacon-chain/rpc/validator/assignments.go

```diff
@@ -58,14 +58,14 @@ func (vs *Server) GetDuties(ctx context.Context, req *ethpb.DutiesRequest) (*eth
 
 		idx, ok := s.ValidatorIndexByPubkey(bytesutil.ToBytes48(pubKey))
 		if ok {
+			assignment.ValidatorIndex = idx
+			assignment.Status = vs.assignmentStatus(idx, s)
+			assignment.ProposerSlots = proposerIndexToSlots[idx]
+
 			ca, ok := committeeAssignments[idx]
 			if ok {
 				assignment.Committee = ca.Committee
-				assignment.Status = vs.assignmentStatus(idx, s)
-				assignment.ValidatorIndex = idx
-				assignment.PublicKey = pubKey
```

Comment:
Are you still assigning `PublicKey`?

Reply:
Yeah, it's set by default on top, look at the collapsed code above.